iroh-resolver/src/racing.rs

```diff
@@ -42,7 +50,10 @@ impl RacingLoader {
     async fn fetch_http(&self, cid: &Cid) -> Result<(Bytes, String), anyhow::Error> {
         let gateway = self.try_raw_gateway()?;
         let cid_str = multibase::encode(multibase::Base::Base32Lower, cid.to_bytes().as_slice());
-        let gateway_url = format!("https://{}.ipfs.{}?format=raw", cid_str, gateway);
+        let mut gateway_url = format!("https://{}.ipfs.{}?format=raw", cid_str, gateway);
```

Comment:
```rust
let gateway_url = if gateway.starts_with("https://") || gateway.starts_with("http://") {
  format!("{}/ipfs/{}?format=raw", gateway, cid_str);
} else {
  format!("https://{}.ipfs.{}?format=raw", cid_str, gateway);
}
```
?

Reply:
oh that's much cleaner. ty!

Reply:
almost everything in rust is an expression. So coming from FP I like htat very much...